From 33aa42d5a50f13dd8b48f1385a7de29aa9010c87 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Tue, 10 Sep 2024 00:10:11 +0800
Subject: [PATCH] fix(evdev-mt-touchpad-tap.c): enable tap-to-click by default

Seriously, it's 2024... This is a partial revert of:

  commit 2219c12c3aa45b80f235e761e87c17fb9ec70eae
  Author: Peter Hutterer <peter.hutterer@who-t.net>
  Date:   Tue Feb 4 10:38:21 2014 +1000

    touchpad: hook up to the tapping configuration

    Now that we have run-time changes of the tap.enabled state move the check
    to the IDLE state only. Otherwise the tap machine may hang if tapping is
    disabled while a gesture is in progress.

    Two basic tests are added to check for the tap default setting - which is now
    "tap disabled by default", for two reasons:
    * if you don't know that tapping is a thing (or enabled by default), you get
      spurious button events that make the desktop feel buggy.
    * if you do know what tapping is and you want it, you usually know where to
      enable it, or at least you can search for it.

    Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
---
 src/evdev-mt-touchpad-tap.c | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/src/evdev-mt-touchpad-tap.c b/src/evdev-mt-touchpad-tap.c
index 299cd554..6e332275 100644
--- a/src/evdev-mt-touchpad-tap.c
+++ b/src/evdev-mt-touchpad-tap.c
@@ -1425,17 +1425,6 @@ tp_tap_default(struct evdev_device *evdev)
 	 */
 	if (!libevdev_has_event_code(evdev->evdev, EV_KEY, BTN_LEFT))
 		return LIBINPUT_CONFIG_TAP_ENABLED;
-
-	/**
-	 * Tapping is disabled by default for two reasons:
-	 * * if you don't know that tapping is a thing (or enabled by
-	 *   default), you get spurious mouse events that make the desktop
-	 *   feel buggy.
-	 * * if you do know what tapping is and you want it, you
-	 *   usually know where to enable it, or at least you can search for
-	 *   it.
-	 */
-	return LIBINPUT_CONFIG_TAP_DISABLED;
 }
 
 static enum libinput_config_tap_state
-- 
2.46.0

